vyper/cli/vyper_compile.py

```diff
@@ -272,6 +313,11 @@ def compile_files(
         output_formats = combined_json_outputs
         show_version = True
 
+    # formats which can only be requested as a single output format
+    for c in ("solc_json", "archive"):
```

Comment:
can't we merge with `combined_json` above?

Reply:
the behavior is slightly different. i had written it both ways and found this was clearer than having a special case inside the loop